yarn2nix.nix

```diff
@@ -26,31 +26,73 @@ let
     main = "main/index.js";
   };
   newPackagePath = builtins.toFile "package.json" (builtins.toJSON newPackage);
+  windowsElectronVersion = "3.0.14";
+  windowsElectron = fetchurl {
+    url = "https://github.com/electron/electron/releases/download/v${windowsElectronVersion}/electron-v${windowsElectronVersion}-win32-x64.zip";
+    sha256 = "0cqwjmv1ymwa309v025szs6681f891s6ks653jd5mh55hp1vpn0b";
+  };
+  checksums = fetchurl {
+    url = "https://github.com/electron/electron/releases/download/v${windowsElectronVersion}/SHASUMS256.txt";
+    sha256 = "103m5kxgb64clx68qqfvxdz2pah249lk344mjxqj94i83v9bxd2j";
+  };
+  electron-cache = runCommand "electron-cache" {} ''
+    mkdir $out
+    ln -s ${windowsElectron} $out/electron-v3.0.14-win32-x64.zip
+    ln -s ${checksums} $out/SHASUMS256.txt-3.0.14
+  '';
+  filter = name: type: let
+    baseName = baseNameOf (toString name);
+    sansPrefix = lib.removePrefix (toString ./.) name;
+  in (
+      baseName == "package.json" ||
+      baseName == "gulpfile.js" ||
+      (lib.hasPrefix "/source" sansPrefix) ||
+      (lib.hasPrefix "/flow" sansPrefix) ||
+      baseName == ".babelrc" ||
+      sansPrefix == "/scripts" ||
+      sansPrefix == "/scripts/package.js" ||
+      sansPrefix == "/installers" ||
+      (lib.hasPrefix "/installers/icons" sansPrefix)
+    );
 in
 yarn2nix.mkYarnPackage {
   name = "daedalus-js";
-  src = if canUseFetchGit then builtins.fetchGit ./. else lib.cleanSource ./.;
+  src = lib.cleanSourceWith { inherit filter; src = ./.; };
   API = api;
   API_VERSION = apiVersion;
   CI = "nix";
   NETWORK = cluster;
   WALLET_PORT = walletPortMap.${cluster};
   BUILD_NUMBER = "${toString buildNum}";
   NODE_ENV = "production";
-  installPhase = ''
+  extraBuildInputs = if win64 then [ wine nukeReferences ] else [ nukeReferences ];
+  installPhase = let
+    nukeAllRefs = ''
+      for x in {main,renderer}/index.js{,.map} main/preload.js{,.map} main/0.js{,.map} renderer/styles.css.map; do
+        nuke-refs $x
```

Comment:
Paths for backtraces -> spurious refs -> extra deps -> we don't want them, so dropping.

Maybe a comment will help here some poor souls : -)